docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,174 @@
+
+## Server-side JavaScript errors
+
+If you encounter unexpected JavaScript errors during the `npm install` or `build` steps, especially errors that other team members can not reproduce, check the versions of Node and npm being used. From a terminal, run:
+```
+node -v
+npm -v
+```
+Note that we test JSS using the Long Term Support (TLS) versions of Node. These are typically one major version behind the latest official Node version.
+
+Note that the Node/npm version used by your CI/Production environments may be different from what is used by your local environment. If a project configuration does not require a specific Node/npm version in `package.json`, deployment agents commonly build using either the newest version available or an environment-specific "default" version.
+
+If you need to swap between multiple Node versions on your local environment for testing, 3rd party packages like [n](https://github.com/tj/n) or [nvm](https://github.com/nvm-sh/nvm) can help with this task.
+
+System Pre-requisites
+Running Node LTS version (use node -v to check)
+
+## Errors regarding SSL certificates
+
+Error examples
+> Error: unable to verify the first certificate
+>
+> Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
+
+> UnauthorizedError: invalid signature
+
+If you are working with a local Sitecore instance using a privately signed certificate, ensure you've [configured Sitecore CA certificates for Node.js](/docs/temp/node-certificates)"
+
+## Errors deploying a JSS app locally
+
+```
+IMPORT ERROR(S) OCCURRED!
+Exception thrown while importing JSS app
+Exception: Sitecore.Exceptions.AccessDeniedException
+Message: AddFromTemplate - Add access required
+```
+
+You will see this type of error when the import role doesn't have the correct permissions. See the following Knowledge Base article for the solution: https://kb.sitecore.net/articles/650791.
+
+## Data-fetching issues
+Building and rendering the sample Next.js app depend on successfully fetching data using GraphQL. The most common cause of failed builds and apps not being able to render pages is an issue with this data-fetching process. The `GraphiQL` interface at `<sitecore hostname + graphQL endpoint>/ui?sc_apikey=<api key>` is a visual browser for GraphQL data. This is a great tool for diagnosing these types of issues.
+
+To support SSG functionality, the sample app exports a `getStaticPaths` function from the <a href="https://github.com/Sitecore/jss/blob/master/samples/nextjs/src/pages/%5B%5B...path%5D%5D.tsx">[[...path]] page</a>, which needs to provide the list of pages to pre-render to Next.js. To check whether `getStaticPaths` is able to get the list of pages, try running the query that `getStaticPaths` uses directly in GraphiQL.
```

Comment:
I found out the "..." in these links need to be encoded (i.e. `%5B%5B%2E%2E%2Epath%5D%5D.tsx`) for them to work (404 otherwise). Seems like they're being encoded as an ellipses??

Reply:
It looks like this is a showdown bug https://github.com/showdownjs/showdown/issues/634